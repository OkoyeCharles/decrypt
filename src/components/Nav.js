import React from 'react';
import '../styles/Nav.css';

function Nav() {
  return (
    <div className="nav">
      <div className="nav__container">
        Hey I&apos;m the nav.
      </div>
    </div>
  );
}

export default Nav;
