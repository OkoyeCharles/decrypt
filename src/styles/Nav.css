.nav {
  position: sticky;
  top: 0;
  background: #fff;
  box-shadow: 0 0 .5em #bbae82;
}

.nav__container {
  padding: .5em;
  display: flex;
  flex-direction: row-reverse;
  align-items: center;
  color: #efb90a;
}

.nav__logo {
  width: 2.5em;
}

.nav__description {
  font-size: 1.25em;
  margin-inline: auto;
}

.nav__back {
  box-shadow: 0 0 0 .05em #efb90a;
  font-size: 1.75em;
  padding: .25em;
  border-radius: 25%;
  margin-right: -1em;
}

.nav__back:active {
  color: white;
  background: #efb90a;
}