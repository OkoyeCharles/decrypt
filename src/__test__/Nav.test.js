import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Nav from '../components/Nav';

it('<Nav /> renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Nav />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
