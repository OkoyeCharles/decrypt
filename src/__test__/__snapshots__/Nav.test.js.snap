// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders correctly 1`] = `
<div
  className="nav"
>
  <div
    className="nav__container"
  >
    <img
      alt=""
      className="nav__logo"
      src="logo.png"
    />
    <div
      className="nav__description"
    >
      Decrypt Stats
    </div>
  </div>
</div>
`;
